src/commands/migration_tasks.py

```diff
@@ -0,0 +1,1358 @@
+                                     f"PORID NOT IN ({porid_exclude_query}) "
+                                     "ORDER BY UPDATED, SECID OFFSET :offset ROWS FETCH NEXT :limit ROWS ONLY;",
+                                     {
+                                         "limit": limit,
+                                         "offset": offset,
+                                         "updated": updated.isoformat(),
+                                         "secid": security.original_id
+                                     })
+
+    @staticmethod
+    def upsert_portfolio_transaction(backend_session: Session,
+                                     portfolio_transaction: destination_models.PortfolioTransaction,
+                                     transaction_number: int,
+                                     transaction_date: date,
+                                     amount: Decimal,
+                                     purchase_c_value: Decimal,
+                                     portfolio_id: UUID,
+                                     security_id: UUID,
+                                     updated: datetime):
+        """
+        Upserts portfolio transaction
+        Args:
+            backend_session: backend database session
+            portfolio_transaction: portfolio transaction
+            transaction_number: transaction number
+            transaction_date: transaction date
+            amount: amount
+            purchase_c_value: purchase c value
+            portfolio_id: portfolio id
+            security_id: security id
+            updated: updated time
+
+        Returns: created / updated entry
+        """
+        new_port_trans = portfolio_transaction if portfolio_transaction else destination_models.PortfolioTransaction()
+        new_port_trans.transaction_number = transaction_number
+        new_port_trans.transaction_date = transaction_date
+        new_port_trans.amount = amount
+        new_port_trans.purchase_c_value = purchase_c_value
+        new_port_trans.portfolio_id = portfolio_id
+        new_port_trans.security_id = security_id
+        new_port_trans.updated = updated
+        backend_session.add(new_port_trans)
+        return new_port_trans
+
+
+class MigrateFundsTask(AbstractMigrationTask):
+    """
+    Migration task for funds
+    """
+
+    def __init__(self):
+        """
+        Constructor
+        """
+        self.kiid_engine = self.get_kiid_engine()
+
+    def get_name(self):
+        return "funds"
+
+    def up_to_date(self, backend_session: Session) -> bool:
+        with Session(self.kiid_engine) as kiid_session:
+            backend_fund_count = backend_session.execute(statement="SELECT COUNT(id) FROM fund").fetchone()
+            kiid_fund_count = kiid_session.execute(statement="SELECT COUNT(ID) FROM FUND").fetchone()
+            return backend_fund_count >= kiid_fund_count
+
+    def migrate(self, backend_session: Session, timeout: datetime) -> int:
```

Comment:
so we are migrating funds from kiid db and the rest from fund db? then why do we have AbstractSalkkuTask? seems not used anywhere.

Reply:
You are correct, AbstractSalkkuTask is no longer used anywhere so I removed it